apps/passport-server/src/services/generic-issuance/pipelines/LemonadePipeline.ts

```diff
@@ -379,6 +380,81 @@ export class LemonadePipeline implements BasePipeline {
     });
   }
 
+  private manualTicketToTicketData(
+    manualTicket: ManualTicket,
+    sempahoreId: string
+  ): ITicketData {
+    const event = this.definition.options.events.find(
+      (event) => event.genericIssuanceEventId === manualTicket.eventId
+    );
+
+    /**
+     * This should never happen, due to validation of the pipeline definition
+     * during parsing. See {@link LemonadePipelineOptionsSchema}.
+     */
+    if (!event) {
+      throw new Error(
+        `Manual ticket specifies non-existent event ID ${manualTicket.eventId} on pipeline ${this.id}`
+      );
+    }
+    const product = event.ticketTypes.find(
+      (product) => product.genericIssuanceProductId === manualTicket.productId
+    );
+    // As above, this should be prevented by pipeline definition validation
+    if (!product) {
+      throw new Error(
+        `Manual ticket specifies non-existent product ID ${manualTicket.productId} on pipeline ${this.id}`
+      );
+    }
+    return {
+      ticketId: manualTicket.id,
+      eventId: manualTicket.eventId,
+      productId: manualTicket.productId,
+      attendeeEmail: manualTicket.attendeeEmail,
+      attendeeName: manualTicket.attendeeName,
+      attendeeSemaphoreId: sempahoreId,
+      isConsumed: false,
+      isRevoked: false,
+      timestampSigned: Date.now(),
+      timestampConsumed: 0,
+      ticketCategory: TicketCategory.Generic,
+      eventName: event.name,
+      ticketName: product.name,
+      checkerEmail: undefined
+    };
+  }
+
+  private async getTicketsForEmail(
```

Comment:
worth mentioning in a comment that this gets both tickets from integration + manual tickets